### DISCUSSION TITLE
[Support/Question] Update livewire model from javascript.

### DISCUSSION THREAD

**Discussion by clarg18** [livewire/livewire#1902] [4. General]
Hello,

I am trying to share the state of a JavaScript variable with Livewire:

This is my Livewire component:

```html
<div>
	...
    <input wire:model="user_code" name="user_code" id="user_code" class="border">
</div>


<script>
    let editor = ace.edit("editor");
    editor.setTheme("ace/theme/monokai");
    editor.session.setMode("ace/mode/php");
    editor.setOptions({
        fontSize: '12pt',
        enableBasicAutocompletion: true,
    });
    editor.session.replace(new ace.Range(0, 0, 1, 1), "<\?php \n");
    editor.gotoLine(2);

    let input = document.getElementById('user_code');
    editor.session.on('change', function(delta) {
        input.value = editor.getValue();
    })
    input.value = editor.getValue();

</script>

```

You can see here, I have an Ace editor (code editor with highlighting etc.) 

I have attempted to bind it's value, which I can retrieve with `editor.getValue()` to a HTML input:

```js
    let input = document.getElementById('user_code');
    editor.session.on('change', function(delta) {
        input.value = editor.getValue();
    })
```

And then put a `wire:model` on that input. 

```html
<input wire:model="user_code" name="user_code" id="user_code" class="border">
```

This does update the HTML input's value but not the wire:model. I have to manual type something extra in the input for it to update the livewire model. 

I tried using alpine.js and @entangle but I could not get that working.

Please can anyone help me achieve this?

**Comment by roni-estein** [comment]
AlpineJs, sorry doing this from my phone in data you can set a holding variable {ml: null} in init run you init config. In the data area I’d add x-on:change=“$set or $wire.whatever” to update your component. You may want that on a different event or you could probably work around it with a wire model lazy. Check out the trix video for some ideas as well.
